### PR TITLE
Make eigen dependency private

### PR DESCRIPTION
**Pull request by IgorBaratta**
should `EIGEN_MAX_ALIGN_BYTES` be private too?
**EDIT**: EIGEN_MAX_ALIGN_BYTES removed.

### CODE REVIEW

**Comment by garth-wells** [comment]
I think we can remove `EIGEN_MAX_ALIGN_BYTES` now and let Eigen do its default thing.

File: python/test/unit/fem/test_assemble_cppimport.py [hunk 1]
```diff
@@ -25,10 +25,11 @@
 def test_eigen_assembly(tempdir):  # noqa: F811
     """Compare assembly into scipy.CSR matrix with PETSc assembly"""
     def compile_eigen_csr_assembler_module():
+        eigen_dir = "/usr/include/eigen3"
```
Comment (garth-wells) [inline python/test/unit/fem/test_assemble_cppimport.py:28]: This will break when Eigen is installed elsewhere. Does Eigen support `pkgconfig`?

**Comment by jhale** [comment]
Forcing the eigen alignment bytes can be removed now. People can use eigen
data structures internally as they please.

File: python/test/unit/fem/test_assemble_cppimport.py (hunk 1 above)
Comment (IgorBaratta) [reply python/test/unit/fem/test_assemble_cppimport.py]: I think so, 
`eigen_dir = dolfinx.pkgconfig.parse("eigen3")["include_dirs"]` provides the eigen include dir.

Comment (garth-wells) [reply python/test/unit/fem/test_assemble_cppimport.py]: My system doesn't have a pkgconfig file for Eigen, so we your elegantly skip the test if `dolfinx.pkgconfig.parse` doesn't find Eigen.

**Comment by IgorBaratta** [comment]
Needs some work on `basix` first.
The header `basix.h` doesn't include eigen, but  the basix library has it on its interface.

File: python/test/unit/fem/test_assemble_cppimport.py (hunk 1 above)
Comment (garth-wells) [reply python/test/unit/fem/test_assemble_cppimport.py]: I found `eigen3.pc` - pkg-config path was messed up.

**Comment by IgorBaratta** [comment]
I think this is now ready for review since `eigen` has been removed from `basix`.

**Review by garth-wells — APPROVED** [review]

**Pull request merged**